Aero/Squadrons: various changes

SJuliez opened the pull request:
This PR makes the following changes:
- No longer shows TMM pips for Aero units
- No longer shows "Internal" in the unit tooltip for fighter squadrons
- Treats customizing a fighter squadron in the lobby as a multi-unit custom, so pilot, quirks etc can no longer be changed
- On the in-game unit sprite, removes the [T] indicator for transporting other units from fighter squadrons as they only transport their own fighters 
- On the in-game unit sprite, removes the internal structure bar for fighter squadrons as they don't suffer SI/Internalstruc damage
- In the lobby, no longer shows a role for fighter squadrons (it was always "undetermined")
- On the minimap of a space map, fades out stars and hex borders somewhat at low zoom (small minimap) so they don't overwhelm the image as much
- Increases randomness for the image choice for the base image of hexes; this removes the stripy-ness of the space map (it is still deterministic and should not change appearance randomly)
- Allows loading fighter squadrons onto carriers; sadly this is still limited to ASF bays that are at least as big as the whole squadron, so for a Leopard CV only squadrons of at most 2 fighters can be loaded; distributing an fsq into multiple bays would require quite a bit more coding as the current transporters don't support that; the ugly part of this change are the additions in GameManager that deal with changing squadrons when they're carried
- Allows calling up the "View" of a fighter squadron in the lobby (some guard code in the TROView)
- And a little fun thing: fsq now construct their icon from the actual fighters:
![image](https://github.com/MegaMek/megamek/assets/17069663/7f665c75-1131-4494-a6c8-6bf1ec6e887e) ![image](https://github.com/MegaMek/megamek/assets/17069663/9f53565a-5917-4ea1-8073-21939577b522) ![image](https://github.com/MegaMek/megamek/assets/17069663/147745f9-bf03-4ccd-869e-0c862d07fa5e) ![image](https://github.com/MegaMek/megamek/assets/17069663/28742db7-36bf-46a9-b31c-8c31b97a8644) ![image](https://github.com/MegaMek/megamek/assets/17069663/ebca67a6-c320-483b-91ab-ab8ee6dab307)

NickAragua commented on megamek/src/megamek/client/ui/swing/boardview/EntitySprite.java:
```diff
@@ -552,8 +552,8 @@ public void prepare() {
-            // Transporting
-            if (!entity.getLoadedUnits().isEmpty()) {
+            // Transporting (but not Squadrons that are obviously composed of subunits)
+            if (!entity.getLoadedUnits().isEmpty() && !(entity instanceof FighterSquadron)) {
```
You may as well use "isSquadron" here.

SJuliez replied:
Yep

NickAragua commented on megamek/src/megamek/client/ui/swing/tileset/HexTileset.java:
```diff
@@ -218,7 +218,8 @@
 
-        Image img = bestMatch.getImage(comp, hex.getCoords().hashCode());
+        Random random = new Random(hex.getCoords().hashCode());
+        Image img = bestMatch.getImage(comp, Math.abs(random.nextInt() * random.nextInt()));
```
Is this null actually possible here?

SJuliez replied:
Only when there are no base images in the tileset or it didnt load, so only when things are very bad. I'd say this can stay as it was (its the bestMatch.getImage() call and used to be like that before)

NickAragua reviewed (changes requested):
Just a little cleanup.

SJuliez commented:
Very sorry, this actually needed a bit more cleanup. I added positioning for squadrons of up to 10 fighters, disallowed icon export for fsq in the lobby with a specific camo (doesnt work with the new icons) and added a static constructor for EntityImage. Aaaah and I'll need to check if this affects MHQ. Not quite there yet!

SJuliez commented:
OK, I added the constructors back in EntityImage for MHQ. Now it should be good (or at least better)
